Fix the no unique key warning for the `<AppSubNav>` component

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

Fix the no unique key warning when viewing the **Reports** page

![2021-11-05 17 05 35](https://user-images.githubusercontent.com/17420811/140486154-3df451fe-a57b-445d-bd6a-d1078983a5f6.png)

### Detailed test instructions:

1. Open browser DevTool.
2. Go to the Reports page.
3. It should no longer show `Warning: Each child in a list should have a unique "key" prop.` message in the console.

### Changelog entry

eason9487 committed:
Fix the no unique key warning for the `<AppSubNav>`

ecgan commented:
@eason9487 , 

> Fix the no unique key warning when viewing the Reports page

I used the `develop` branch (commit 064ab39147e74dd09c5759ac8b0982d159c3b38d), I visited the Reports page `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Freports`, and clicked on the "Programs" and "Products" sub tab, and I don't see the warning in console? 🤔 

![image](https://user-images.githubusercontent.com/417342/140565246-eac6327d-cd36-45a4-83f8-b37fb97975fd.png)

ecgan commented on js/src/components/app-sub-nav/index.js:
```diff
@@ -33,7 +34,7 @@ const AppSubNav = ( props ) => {
-					<>
+					<Fragment key={ tab.key }>
 						<Link
 							key={ tab.key }
```
This is not needed anymore, we can remove this. 🙂 

```suggestion
```

ecgan reviewed (approved):
Tested and working okay, but I'm also wondering why I don't see the issue when I checkout `develop` branch, as I commented in https://github.com/woocommerce/google-listings-and-ads/pull/1079#issuecomment-962148465. Left one suggestion below. LGTM 👍

eason9487 committed:
Remove an unneeded prop in AppSubNav

Address https://github.com/woocommerce/google-listings-and-ads/pull/1079#discussion_r743918963

Co-authored-by: Gan Eng Chin <ecgan@users.noreply.github.com>

eason9487 commented:
@ecgan,

> I'm also wondering why I don't see the issue when I checkout `develop` branch

Add the below to the *wp-config.php* file.
```php
// Use dev versions of core JS and CSS files
define( 'SCRIPT_DEBUG', true );
```

ecgan commented:
@eason9487 , thanks for that [`SCRIPT_DEBUG`](https://wordpress.org/support/article/debugging-in-wordpress/#script_debug) info, it works. 🙂 👍
